core/tests/performance_tests/string_classifier_performance_test.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Performance tests for string classifier.
+
+Run this script from the Oppia root directory:
+
+    export PYTHONPATH=$PYTHONPATH:.
+    python core/tests/performance_tests/string_classifier_performance_test.py
+
+"""
+
+import os
+import time
+import yaml
+
+from core.domain.classifier_services import StringClassifier
+
+def benchmark(func):
+    """ A decorator to measure how much time spent by func
+
+    Args:
+        func: function. The function to be measured.
+
+    Returns:
+        function. The decorated function by this decorator.
+    """
+    def time_taken(obj, num):
+        """ Measures the time spent by func.
+
+        Args:
+            obj: object. The class instance func belongs to.
+            num: int. The number of samples to run with.
+
+        Returns:
+            Result of func.
+        """
+        start = time.time()
+        result = func(obj, num)
+        end = time.time()
+        print '%s spent %f seconds for %d instances' % (func.__name__,
+                                                        end - start, num)
+        return result
+    return time_taken
+
+class StringClassifierBenchmarker(object):
+    """ Benchmark for string classifier.
+
+    Usage:
+        benchmark = StringClassifierBenchmark()
+        benchmark.benchmark_on_train()
+        benchmark.benchmark_on_predict()
+
+    Attributes:
+        examples: list of two-element lists. An item represents a sample
+            by [doc, label].
+        predict_docs: list. A list of docs for predicting.
```

Comment:
This is list(str)?

Also, consider: docs_to_classify or prediction_docs or test_data or something that is more like a noun. Starting a variable name with "predict" suggests that it is a function/method.